### ISSUE TITLE
Ship native darwin arm64 binaries with the macOS arm64 version

### ISSUE THREAD

**Issue by dvaldivia** [rancher-sandbox/rancher-desktop#4564]
### Problem Description

Most of the binaries in the rancher distribution are x86 causing macOS to rely on `qemu-system-aarch64` which uses plenty of CPU and drains the battery faster, I suspect the internal VM is also x86, however by bundling native arm64 binaries with rancher the problem can start getting mitigated

inside `Rancher Desktop.app/Contents/Resources/resources/darwin/bin` 
```
./docker: Mach-O 64-bit executable arm64
./docker-buildx: Mach-O 64-bit executable arm64
./docker-compose: Mach-O 64-bit executable arm64
./docker-credential-ecr-login: Mach-O 64-bit executable arm64
./docker-credential-none: Mach-O 64-bit executable x86_64
./docker-credential-osxkeychain: Mach-O 64-bit executable arm64
./helm: Mach-O 64-bit executable arm64
./kubectl: Mach-O 64-bit executable x86_64
./kuberlr: Mach-O 64-bit executable x86_64
./nerdctl: Bourne-Again shell script text executable, ASCII text
./rdctl: Mach-O 64-bit executable x86_64
```

Inside `Rancher Desktop.app/Contents/MacOS`

```
Rancher Desktop: Mach-O 64-bit executable x86_64
```

### Proposed Solution

Bundle arm64 binaries

**Comment by jandubois** [comment]
> Most of the binaries in the rancher distribution are x86 causing macOS to rely on `qemu-system-aarch64` which uses plenty of CPU and drains the battery faster

I don't understand what you are saying. `qemu-system-aarch64` is the native binary for `arm64`. It is the code that runs the virtual Linux machine.

> I suspect the internal VM is also x86

That is not correct. As you noticed, it is being run by `qemu-system-aarch64`.


> ```
> ./docker-credential-none: Mach-O 64-bit executable x86_64
> ./kubectl: Mach-O 64-bit executable x86_64
> ./kuberlr: Mach-O 64-bit executable x86_64
> ./rdctl: Mach-O 64-bit executable x86_64
> ```

These are the only x86_64 binaries (except for the GUI Electron app itself), and none are performance-critical.

The reason `kuberlr` and `kubectl` use `x86_64` binaries is that upstream releases only exist for k8s 1.21.0 and later, so we wouldn't be able to support older k8s versions on ARM otherwise.

I'm surprised the other 2 CLI tools are `x86_64` versions, but again, they are not performance-critical at all.

This is a duplicate of #1416, but I'll keep this issue open until we have determined why `rdctl` and `docker-credential-none` are not native binaries.

**Comment by fulldecent** [comment]
Got it, so we need to follow that issue if we are waiting on a complete, no Rosetta, M1 version of Rancher. ➡️

**Comment by jandubois** [comment]
> Got it, so we need to follow that issue if we are waiting on a complete, no Rosetta, M1 version of Rancher. ➡️

Yes. Note that you can install your own aarch64 version of `kubectl`, and make sure it is in a directory at the front of your `PATH`. It is then up to you to make sure that version is compatible with the Kubernetes cluster version you are running. Or you can even install the aarch64 version of `kuberlr` as long as you don't use clusters older than 1.21.0.

The reason we are using the x86_64 version of `kuberlr` is because it can't download pre-1.21.0 versions of aarch64 `kubectl` because the upstream project never published them.
